corehq/apps/userreports/reports/data_source.py

```diff
@@ -123,22 +122,38 @@ def get_data(self, slugs=None):
             if self._order_by:
                 for col in reversed(self._order_by):
                     is_descending = col[1] == DESCENDING
-                    is_date = any(
-                        configured_indicator['datatype'] == 'date'
-                        for configured_indicator in self.config.configured_indicators
-                        if configured_indicator['column_id'] == col[0]
-                    )
-                    default_sort_by_date = (
-                        date(datetime.MINYEAR, 1, 1)
-                        if is_descending else date(datetime.MAXYEAR, 12, 31)
-                    )
-                    value = lambda x: x.get(col[0], None)
-                    sort_by_value = lambda x: (
-                        value(x)
-                        or (default_sort_by_date if is_date else value(x))
-                    )
+
+                    def sort_by(row):
+                        value = row.get(col[0], None)
+                        if not value:
+                            matching_indicators = filter(
+                                lambda configured_indicator: configured_indicator['column_id'] == col[0],
+                                self.config.configured_indicators
+                            )
+                            if not len(matching_indicators) == 1:
+                                raise UserReportsError(
+                                    'Number of indicators matching column %(col)s is %(num_matching)d' % {
+                                        'col': col[0],
+                                        'num_matching': len(matching_indicators),
+                                    }
+                                )
+                            if matching_indicators[0]['datatype'] == 'date':
```

Comment:
not actionable, but one way i like to structure code like this is to build a dict that is kind of like a branch tree and then just call it. something like:
```python
defaults = { 
  "date": {
    "max": datetime.date(datetime.MAXYEAR, 12, 31)
    "min": datetime.date(datetime.MINYEAR, 1, 1)
  } 
  "datetime" { // etc
```
and then the calling code can be:
```python
defaults.get(matching_indicators[0]['datatype'])['max' if is_descending else 'min']
```
you need to use `defaultdict`s or have missing key handling, but i find it makes the intention and code cleaner